c10/core/TensorOptions.cpp

```diff
@@ -9,6 +9,16 @@
 
 namespace c10 {
 
+thread_local bool NonVariableTypeMode_enabled = false;
```

Comment:
Mobile in Facebook is going to throw a hissy-fit about the thread local, unless something has changed since I last looked. You'll need some test like what was removed in 5c1692840e423e43e12ba8346c46bf121d9520c6

```
-// In the CAFFE2_FB_LIMITED_MOBILE_CAPABILITY build setting,
-// thread_local is not supported.  In that case, we don't provide
-// an OptionsGuard; and force you to pass around options manually.
-#if !C10_MOBILE && !defined(CAFFE2_FB_LIMITED_MOBILE_CAPABILITY)
```

Reply:
There should be comments here describing the stuff that is inside the design proposal. Personally, I'd make it a Note, and cross-ref it from the `is_variable()` implementation

Reply:
@ezyang Since the thread-local guard is used by the dispatching in VariableType.cpp, is it safe to assume that mobile is not going to depend on libtorch and/or use variable type dispatch?

Reply:
Yeah, if you're trying to decide what to stub, just having it not work at all on mobile is probably OK. We should revisit if people start using this guard in other places.